### PR TITLE
Dx 36 cli developer is able to review their plugins information details

### PR DESCRIPTION
**Pull request by pythonpete32**
- adds info command: displays information about plugin repos
added version info

[command in action](https://share.cleanshot.com/Y4VjCf09)

### CODE REVIEW

File: src/commands/info/infoHandler.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const pluginRepos = await queryPluginRepos(network.subgraph);
+  const plugin = repoName ? findRepo(repoName, pluginRepos) : await pluginSelectionPrompt(pluginRepos);
+
+  const releaseMetadata = await downloadFromIPFS(plugin.releases[0].metadata);
+
+  logTable([
+    { NAME: releaseMetadata.name || strings.NO_NAME_PROVIDED },
+    { SUBDOMAIN: plugin.subdomain },
+    { DESCRIPTION: releaseMetadata.description || strings.NO_DESCRIPTION_PROVIDED },
+    { ADDRESS: plugin.id },
```
Comment (juliettech13) [inline src/commands/info/infoHandler.ts:26]: Shall we add maintainer here as well?

File: src/commands/publish/publishHandler.ts [hunk 2]
```diff
@@ -37,15 +39,18 @@ export const publishHandler: (...args: any[]) => void | Promise<void> = async (
...
 
-  console.table({
-    setupContract,
-    subdomain: `${subdomain}.plugin.dao.eth`,
-    chain: network.name,
-    maintainer,
-  });
+  logTable([
+    { setupContract },
+    { subdomain: `${subdomain}.plugin.dao.eth` },
```
Comment (juliettech13) [inline src/commands/publish/publishHandler.ts:44]: This is tricky because - as far as I'm concerned - subdomains only work on mainnet. Let's double check here with OS though and fix if needed

File: src/lib/constants.ts [hunk 3]
```diff
@@ -9,25 +10,46 @@ export const exitWithMessage = (message: string): void => {
   process.exit(0);
 };
 
+/**
```
Comment (juliettech13) [inline src/lib/constants.ts:13]: This file is getting convoluted. I'd move this to another file so it lives on its own and keep this file only for strings

File: src/lib/constants.ts [hunk 4]
```diff
@@ -9,25 +10,46 @@ export const exitWithMessage = (message: string): void => {
...
+  const table = new Table();
+  keyValues.forEach((kv) => table.push(kv));
+  console.log(table.toString());
+};
+
 export const strings = {
   ABORTED: error('Aborted.'),
   BUILD_FOLDER: success(`Build Folder not detected. \nPlease enter the name of the build folder:`),
   BUILD_METADATA_PROMPT: `Select your ${chalk.blue('Build metadata')} file:`,
+  CANNOT_FIND_REPO: (repoName: string) => error(`Cannot find repo ${repoName}`),
```
Comment (juliettech13) [inline src/lib/constants.ts:31]: When does this come up?

File: src/lib/constants.ts [hunk 5]
```diff
@@ -9,25 +10,46 @@ export const exitWithMessage = (message: string): void => {
...
   ENABLED: chalk.green('enabled'),
   EXPLORER: (explorer: string, txHash: string | undefined) => success(`🔗 ${explorer}/tx/${txHash}`),
   FAILED_TO_RETRIEVE_KEYS: error('Failed to retrieve keys:'),
+  IPFS_NOT_VALID_JSON: error('IPFS content is not valid JSON'),
   IPFS_UPLOAD_ERROR: error('Failed to upload to IPFS:'),
   INVALID_BUILD_METADATA: error('Invalid build metadata'),
   INVALID_RELEASE_METADATA: error('Invalid release metadata'),
   INVALID_SUBDOMAIN: error('Invalid subdomain'),
   MAINTAINER_ADDRESS: success("Please enter a maintainer's address:"),
+  METADATA_DOWNLOADED: success('Metadata Downloaded'),
```
Comment (juliettech13) [inline src/lib/constants.ts:48]: Why is the D capitalized?

File: src/lib/constants.ts [hunk 6]
```diff
@@ -55,23 +82,30 @@ export const networks: Array<Network> = [
...
   {
     name: 'mumbai',
     id: '80001',
     url: 'https://rpc.ankr.com/polygon_mumbai',
     explorer: 'https://mumbai.polygonscan.com',
+    subgraph: 'https://subgraph.satsuma-prod.com/qHR2wGfc5RLi6/aragon/osx-mumbai/version/v1.1.1/api',
   },
 ];
+
+export const WEB_3_STORAGE =
```
Comment (juliettech13) [inline src/lib/constants.ts:110]: Why is this here instead of in `.env`?

**Comment by juliettech13** [comment]
In the video I see that some manual prompts are capitalized and others are not. Let's make sure we capitalize all of them.

Aside from that, LOVE all of the feeling + errors! Great job.

File: src/commands/info/index.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { Command, Option } from 'commander';
+import { infoHandler } from './infoHandler';
+
+export const info = new Command('info');
+
+info
+  .description('information about a Plugin Repo')
```
[the remark this answers is not in the document]
Comment (pythonpete32) [reply src/commands/info/index.ts:7]: This make a lot of sense. Thinking back to it, this is something that confused me quite a bit too

File: src/commands/info/infoHandler.ts (hunk 1 above)
Comment (pythonpete32) [reply src/commands/info/infoHandler.ts:26]: yes we should! The problem is the maintainer/s are not being indexed in the subgraph and AFAIK you cant query the `PermissionManager` for a list of addresses with a particular permission (unless im missing something). So we can do it but its alot of work

File: src/commands/publish/publishHandler.ts (hunk 2 above)
Comment (pythonpete32) [reply src/commands/publish/publishHandler.ts:44]: hmmm what do you mean by they only work on mainnet? The canonical ENS deployment is on mainnet but we deploy our own version on the other networks we are on

File: src/lib/constants.ts (hunk 4 above)
Comment (pythonpete32) [reply src/lib/constants.ts:31]: if the user passes the repo without going through the prompt but the repo is not found

File: src/commands/info/infoHandler.ts (hunk 1 above)
Comment (juliettech13) [reply src/commands/info/infoHandler.ts:26]: ok cooool. let's leave it as is for now and chat with the OS team to provide this on their end.

File: src/commands/publish/publishHandler.ts (hunk 2 above)
Comment (juliettech13) [reply src/commands/publish/publishHandler.ts:44]: I don't think we do actually.. let's confirm this with OS.

Also, even if we do, there's no way for devs to use this address and have it resolve to their plugin, so it's a tricky UX.

I suggest we add some copy regarding this, potentially at the bottom after the table shows up - sharing something like: "Keep in mind that your plugin's subdomain will only be resolved in mainnet, as is the only network currently supported by ENS"

Comment (pythonpete32) [reply src/commands/publish/publishHandler.ts:45]: ive added it but if happy to just remove the field if it makes it clearer

Comment (juliettech13) [reply src/commands/publish/publishHandler.ts:45]: I'd add a conditional statement here -  this should only come up if the network supports subdomains. If the network doesn't, then the value for `subdomain` should be: "%{network.name} does not yet support ENS subdomains."

File: src/lib/ipfs.ts [hunk 8]
```diff
@@ -2,7 +2,9 @@ import { BytesLike, ethers } from 'ethers';
 import { Web3Storage, File } from 'web3.storage';
 import { Buffer } from 'buffer';
 import { updateSpinnerText, spinnerSuccess } from './spinners';
-import { exitWithMessage, strings } from './constants';
+// import { WEB_3_STORAGE } from './constants';
```
Comment (juliettech13) [inline src/lib/ipfs.ts:5]: Remove comment

**Comment by juliettech13** [comment]
Missing tests for `info` command

**Review by juliettech13 — APPROVED** [review]
